src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs

```diff
@@ -238,6 +239,15 @@ private static unsafe MsQuicSafeHandle Create(QuicConnectionOptions options, QUI
             {
                 ThrowHelper.ThrowIfMsQuicError(status, SR.net_quic_tls_version_notsupported);
             }
+
+            if (status == MsQuic.QUIC_STATUS_CERT_NO_CERT && certificate != null && certificate.HasPrivateKey())
+            {
+                using Microsoft.Win32.SafeHandles.SafeCertContextHandle safeCertContextHandle = Interop.Crypt32.CertDuplicateCertificateContext(certificate.Handle);
```

Comment:
Why do we need to duplicate?

Reply:
This is the same logic as in SslStream. I assume the reason for that is that the Interop stub for `CertGetCertificateContextProperty` takes `SafeCertContextHandle`, but the `X509Certificate2.Handle` property returns an `IntPtr`. This is probably the easiest path to call that function, given that it's on slow/exceptional path anyway.